core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -18,23 +18,37 @@
  */
 package org.apache.iceberg;
 
+import java.util.List;
 import java.util.Map;
-import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.CatalogWithViews;
+import org.apache.iceberg.catalog.Namespace;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.exceptions.CommitFailedException;
 import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.metrics.MetricsReporter;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.view.BaseView;
+import org.apache.iceberg.view.ImmutableSQLViewRepresentation;
+import org.apache.iceberg.view.ImmutableViewHistoryEntry;
+import org.apache.iceberg.view.ImmutableViewMetadata;
+import org.apache.iceberg.view.ImmutableViewVersion;
+import org.apache.iceberg.view.View;
+import org.apache.iceberg.view.ViewBuilder;
+import org.apache.iceberg.view.ViewMetadata;
+import org.apache.iceberg.view.ViewOperations;
+import org.apache.iceberg.view.ViewRepresentation;
+import org.apache.iceberg.view.ViewVersion;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class BaseMetastoreCatalog implements Catalog {
+public abstract class BaseMetastoreCatalog implements CatalogWithViews {
```

Comment:
what's the difference of this vs `implements Catalog, ViewCatalog`?

Reply:
Both `Catalog` and `ViewCatalog` define `name()` and `initialize(..)` methods and the main purpose of `CatalogWithViews` is to de-conflict this. For `BaseMetastoreCatalog` we can de-conflict this inside `BaseMetastoreCatalog` but there's no way to de-conflict this for `ViewCatalogTests` because we can't define `ViewCatalogTests<C extends Catalog & SupportsNamespaces & ViewCatalog>` without errors

Reply:
I'm not a big fan of making API changes just to facilitate tests. Can we use `instanceof` checks to have multiple ways of interacting with the catalog instead?

Reply:
@rdblue I'm not sure I understand what you mean by using `instanceof` checks to have multiple ways of interacting with the catalog. Prior to  https://github.com/apache/iceberg/pull/7880/commits/fa3ce01275e890fd639b53ee19ab360bf35e4e22 there was a `BaseMetastoreViewCatalog` implementation that had `extends BaseMetastoreCatalog implements ViewCatalog`. In the `ViewCatalogTests` we'd then define a `Catalog catalog()` and a `ViewCatalog viewCatalog()` method to interact with one or the other catalog